src/__rt/future.rs

```diff
@@ -0,0 +1,51 @@
+use std::mem;
+use std::ops::{Generator, GeneratorState};
+
+use super::{IsResult, Mu, Reset, CTX};
+
+use futures::task;
+use futures::prelude::{Poll, Async, Future};
+
+pub trait MyFuture<T: IsResult>: Future<Item=T::Ok, Error = T::Err> {}
+
+impl<F, T> MyFuture<T> for F
+    where F: Future<Item = T::Ok, Error = T::Err > + ?Sized,
+          T: IsResult
+{}
+
+/// Small shim to translate from a generator to a future.
+///
+/// This is the translation layer from the generator/coroutine protocol to
+/// the futures protocol.
+struct GenFuture<T>(T);
+
+pub fn gen_move<T>(gen: T) -> impl MyFuture<T::Return>
+    where T: Generator<Yield = Async<Mu>>,
+          T::Return: IsResult,
+{
+    GenFuture(gen)
+}
+
+impl<T> Future for GenFuture<T>
+    where T: Generator<Yield = Async<Mu>>,
+          T::Return: IsResult,
+{
+    type Item = <T::Return as IsResult>::Ok;
+    type Error = <T::Return as IsResult>::Err;
+
+    fn poll(&mut self, ctx: &mut task::Context) -> Poll<Self::Item, Self::Error> {
+        CTX.with(|cell| {
+            let _r = Reset(cell.get(), cell);
+            cell.set(unsafe { mem::transmute(ctx) });
```

Comment:
Nit: can you replace `transmute` with `as *mut _ as *mut task::Context<'static>`?

Reply:
I can't extend the lifetime of Context to 'static with as casts.

Reply:
Ah whoops, I skipped an `as`. Here's [a playground.](https://play.rust-lang.org/?gist=91093dca6aa5682c3ccaa8d86186f4db&version=stable)

Reply:
In any case, the transmute is probably cleaner. Can you turbofish to make the types explicitly so its clear what transformation you're making? Also, with the refactoring I suggested you should be able to get down to just a single call to `transmute.